src/discord.cpp

```diff
@@ -20,30 +43,60 @@ Discord::Discord(QObject *parent) : QObject(parent)
 
     Discord_Initialize(APPLICATION_ID, &handlers, 1, "1234");
 
-    // start Discord's event loop at 50ms
+    // process Discord callbacks every 50ms
     startTimer(50);
 }
 
 
 Discord::~Discord()
 {
-    Discord_Shutdown();
+    //    Discord_Shutdown();
+}
+
+void Discord::setGameName(const QString& name)
+{
+    if (!mLoaded) {
+        return;
+    }
+
+    char buffer[256];
+    sprintf(buffer, "Playing %s", name.toUtf8().constData());
+    mDiscordPresence.details = buffer;
+    mDiscordPresence.largeImageKey = name.toLower().toUtf8().constData();
+
+    Discord_UpdatePresence(&mDiscordPresence);
+}
+
+void Discord::setStatus(const QString& status)
+{
+    if (!mLoaded) {
+        return;
+    }
+
+    char buffer[256];
+    sprintf(buffer, "%s", status.toUtf8().constData());
+    mDiscordPresence.state = buffer;
+
+    Discord_UpdatePresence(&mDiscordPresence);
 }
 
-void Discord::timerEvent(QTimerEvent *event)
+void Discord::timerEvent(QTimerEvent* event)
 {
+    if (!mLoaded) {
```

Comment:
Is, aborting on fail always the way to go with short methods like this, would not:

    if (mLoaded) {
        Discord_RunCallbacks();
    }

be just as functional but even briefer?

Reply:
Yeah :)